src/saltext/vmware/states/vmc_security_groups.py

```diff
@@ -0,0 +1,421 @@
+"""
+VMC Security group state module
+
+Add new security group, update existing security group and delete existing security group from an SDDC.
+
+Example usage :
+
+.. code-block:: yaml
+
+    ensure_security_group:
+      vmc_security_groups.present:
+        - hostname: sample-nsx.vmwarevmc.com
+        - refresh_key: 7jPSGSZpCa8e5Ouks4UY5cZyOtynAhF
+        - authorization_host: console-stg.cloud.vmware.com
+        - org_id: 10e1092f-51d0-473a-80f8-137652c39fd0
+        - sddc_id: b43da080-2626-f64c-88e8-7f31d9d2c306
+        - domain_id: mgw
+        - security_group_id: Security_group_1
+        - verify_ssl: False
+        - cert: /path/to/client/certificate
+
+.. warning::
+
+    It is recommended to pass the VMC authentication details using Pillars rather than specifying as plain text in SLS
+    files.
+
+
+"""
+import logging
+
+from saltext.vmware.utils import vmc_constants
+from saltext.vmware.utils import vmc_state
+
+log = logging.getLogger(__name__)
+SECURITY_GROUP_NOT_FOUND_ERROR = "could not be found"
+
+
+try:
+    from saltext.vmware.modules import vmc_security_groups
+
+    HAS_SECURITY_GROUPS = True
+except ImportError:
+    HAS_SECURITY_GROUPS = False
+
+
+def __virtual__():
+    if not HAS_SECURITY_GROUPS:
+        return False, "'vmc_security_groups' binary not found on system"
+    return "vmc_security_groups"
+
+
+def present(
+    name,
```

Comment:
**suggestion** as in another review, `name` should be the primary required piece. For this I think it should be the `security_group_id` -- in other words, remove the security_group_id in the function signature, and then `security_group_id = name` as the first line in the function should work here. Obviously tests will need to be updated as well.

Reply:
updated